`ultralytics 8.0.53` DDP AMP and Edge TPU fixes

glenn-jocher opened the pull request:
<!--
Thank you 🙏 for submitting a Pull Request to an Ultralytics 🚀 repo! We want to make contributing as possible. A few tips to get you started:

- Search existing PRs to see if a similar contribution already exists.
- Link this PR to an open Issue to help us understand what bug fix or feature is being implemented.
- Sign the Ultralytics Contributor License Agreement (CLA) by writing the below statement in a PR comment:  
I have read the CLA Document and I sign the CLA

Please see our ✅ [Contributing Guide](https://github.com/ultralytics/ultralytics/blob/master/CONTRIBUTING.md) for more details.
-->


## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Enhancements to GitHub workflows and README updates.

### 📊 Key Changes
- Increased hard-fail threshold in Continuous Integration (CI) from 0.60 to 0.61.
- Added `PERSONAL_ACCESS_TOKEN` as an environment variable for documentation deployment.
- Adjusted contributor images in README files to control width.
- Improved alignment and layout of application documentation and removed unnecessary line breaks.
- Removed old configuration documentation `cfg.md`.
- Added new benchmark documentation `benchmark.md`.
- Created new export documentation `export.md`.
- Structured documentation to include new sections for modes, tasks, and usage.

### 🎯 Purpose & Impact
- 🎈 The harder CI benchmark aims to ensure higher code quality.
- 🏗️ Utilizing the environment variable provides more security and flexibility in CI/CD processes.
- 📐 Adjusting image sizes in READMEs leads to a more consistent and tidy display.
- 📝 Updating documentation sections ensures a better organization and enhances the usability of docs for users.
- 🗑️ Removing outdated documentation minimizes confusion and maintains the clarity of the information.

glenn-jocher committed:
Docs deployment fix

glenn-jocher committed:
macos and windows CI

glenn-jocher committed:
Update CoreML cls models to hard fail 0.58

glenn-jocher committed:
CI updates

glenn-jocher committed:
CI updates

AyushExel commented:
@glenn-jocher can you look into this and incorporate a fix in this release if necessary https://github.com/ultralytics/ultralytics/issues/1367

glenn-jocher committed:
AMP Check only on RANK 0, -1

glenn-jocher committed:
AMP Check only on RANK 0, -1

glenn-jocher committed:
update to parenthesis

glenn-jocher committed:
Remove TF.js on Windows attempted export

glenn-jocher committed:
8.0.53

35grain committed:
Tensor input is expected to be int8 (#1361)

Co-authored-by: Glenn Jocher <glenn.jocher@ultralytics.com>

glenn-jocher committed:
Remove ImageNet normalization for classify models (#1368)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

hotfur committed:
Fix bbox prediction for torch.Tensor source (#1373)

Co-authored-by: Glenn Jocher <glenn.jocher@ultralytics.com>

glenn-jocher committed:
Update cls threshold

glenn-jocher committed:
Cleanup AMP check

glenn-jocher committed:
Update mkdocs navigation

glenn-jocher committed:
Add Tasks index with keypoints placeholder

glenn-jocher commented:
@AyushExel @Laughing-q I've added docs top-level indexing now in https://github.com/ultralytics/ultralytics/pull/1362/commits/59b0d33f44b7fc191da48d63c22d77825d79781c so we can add index.md to subdirectories. I've created a tasks/index.md which explains all the different tasks on one page. I added keypoints to this and also created an empty tasks/keypoints.md placeholder for our other PR.

<img width="1626" alt="Screenshot 2023-03-11 at 19 38 20" src="https://user-images.githubusercontent.com/26833433/224505913-d7241665-fb5e-491a-a433-dc0ddb176956.png">

glenn-jocher committed:
Add keypoints.md content

glenn-jocher committed:
Add keypoints.md content

glenn-jocher committed:
revert onnx2tf changes

glenn-jocher committed:
Add modes to docs

glenn-jocher committed:
Cleanup benchmarks comments

glenn-jocher committed:
Fix val cls augmentation bug

glenn-jocher committed:
Fix val cls augmentation bug

glenn-jocher committed:
Add export args

glenn-jocher committed:
Add valid predict source formats

glenn-jocher committed:
Add valid predict source formats

glenn-jocher committed:
Reduce cls threshold to 0.61

glenn-jocher committed:
Docs update

glenn-jocher committed:
Reformat docs

glenn-jocher committed:
Simplify task names

glenn-jocher committed:
create usage top level directory

glenn-jocher committed:
Update HUB and App pages

glenn-jocher committed:
Update contributors image to 100% width

glenn-jocher committed:
remove macOS benchmarks

glenn-jocher committed:
fix AMP check

glenn-jocher committed:
Add AMP check to tests

glenn-jocher committed:
Comment keypoints task in Docs

glenn-jocher committed:
Update tests

glenn-jocher committed:
Remove forced test

glenn-jocher committed:
AMP fix test

glenn-jocher committed:
attempt broadcast fix

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

glenn-jocher committed:
attempt broadcast fix

glenn-jocher committed:
Merge remote-tracking branch 'origin/updates' into updates

# Conflicts:
#	ultralytics/yolo/engine/trainer.py

glenn-jocher committed:
attempt broadcast fix

glenn-jocher committed:
attempt broadcast fix

glenn-jocher committed:
attempt broadcast fix

glenn-jocher committed:
attempt broadcast fix

glenn-jocher committed:
attempt broadcast fix

glenn-jocher committed:
attempt broadcast fix

glenn-jocher committed:
attempt broadcast fix
